torax/sources/external_current_source.py

```diff
@@ -227,16 +231,28 @@ def jext_hires(
     """Return the external current density profile along the hires cell grid."""
     assert isinstance(dynamic_source_runtime_params, DynamicRuntimeParams)
     self.check_mode(dynamic_source_runtime_params.mode)
+
+    # Interpolate prescribed values onto the hires grid
+    if dynamic_source_runtime_params.mode == runtime_params_lib.Mode.PRESCRIBED:
+      hires_prescribed_values = jnp.interp(
+        geo.r_hires_norm,
+        geo.r_norm,
+        dynamic_source_runtime_params.prescribed_values
```

Comment:
this code block is actually not jax-compatible, meaning if you were to change the `Mode` to/from prescribed, the original code path would still be followed.

use `jnp.where` instead of an if-else

Reply:
Resolved in 8e32310c1220e9316e38dfe54c380581aec388d8